cocktails9/cocktails9/Views/Cocktails/CocktailsGridView.swift

```diff
@@ -0,0 +1,162 @@
+import Foundation
+import SwiftUI
+
+struct CocktailsGridView: View {
+    
+    @StateObject private var viewModel: CocktailViewModel
+    @StateObject private var viewDetailModel: CocktailDetailsViewModel
+    @StateObject private var filterViewModel: FiltersViewModel
+    private let cocktailService: CocktailService
+    private let filterService: FilterService
+    @State private var isFilterPresented = false
+    @State private var searchText = ""
+    @State private var isSearchVisible = false
+    @State private var isLoading = true
+    
+    init(cocktailService: CocktailService, filterService: FilterService) {
+        self.cocktailService = cocktailService
+        self.filterService = filterService
+        _viewModel = StateObject(wrappedValue: CocktailViewModel(cocktailService: cocktailService))
+        _filterViewModel = StateObject(wrappedValue: FiltersViewModel(filterService: filterService))
+        _viewDetailModel = StateObject(wrappedValue: CocktailDetailsViewModel(cocktailService: cocktailService))
+    }
+
+    private let columns: [GridItem] = [
+        GridItem(.flexible(), spacing: 10),
+        GridItem(.flexible(), spacing: 10)
+    ]
+
+    var body: some View {
+        NavigationView {
+            VStack(spacing: 10) {
+                
+                // Search and Filter Bar
+                VStack(spacing: 10) {
+                    HStack(spacing: 10) {
+                        Text("cocktails9")
+                            .font(.largeTitle)
+                            .fontWeight(.bold)
+                            .foregroundColor(.black)
+                            .padding(.leading, 10)
+                        
+                        Spacer()
+                        
+                        // Search Button
+                        Button(action: {
+                            withAnimation {
+                                isSearchVisible.toggle()
+                            }
+                        }) {
+                            Image(systemName: "magnifyingglass")
+                                .padding(10)
+                                .background(Color(.systemGray6))
+                                .cornerRadius(10)
```

Comment:
`.background` modifier is more powerful than you think 🙂 
```suggestion
                                .background(Color.red, in: RoundedRectangle(cornerRadius: 10, style: .circular))
```
so, by using this, you could remove all things from below that is using custom `cornerRadius` func